modules/packages/LinearAlgebra.chpl

```diff
@@ -928,11 +929,14 @@ private proc _diag_vec(A:[?Adom] ?eltType, k) {
     if m < K then halt("k is out of range");
 
     var length = min(n, m - K);
-    const space = dim.first..#length;
-    var diagonal = Vector(space, eltType);
-
-    forall i in space do
-      diagonal[i] = A[i+K, i];
+    var diagonal = Vector(0..#length, eltType);
+    const offset = Adom.dim(0).stride * K;
+
+    forall (i, j, diagInd) in 
```

Comment:
nit: this indentation style reads weirdly for me. How about we match what we did up on line 899?